crates/sui-framework/packages/sui-framework/sources/bridge/bridge.move

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[allow(unused_const)]
+module sui::bridge {
+    use std::vector;
+
+    use sui::address;
+    use sui::balance;
+    use sui::bcs;
+    use sui::bridge_committee::{Self, BridgeCommittee};
+    use sui::bridge_escrow::{Self, BridgeEscrow};
+    use sui::bridge_treasury::{Self, BridgeTreasury, token_id};
+    use sui::chain_ids;
+    use sui::coin::{Self, Coin};
+    use sui::event::emit;
+    use sui::object::{Self, UID};
+    use sui::table::{Self, Table};
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+
+    #[test_only]
+    use sui::bridge_treasury::USDC;
+    #[test_only]
+    use sui::hex;
+    #[test_only]
+    use sui::test_scenario;
+
+    struct Bridge has key {
+        id: UID,
+        // nonce for replay protection
+        sequence_num: u64,
+        // committee pub keys
+        committee: BridgeCommittee,
+        // Escrow for storing native tokens
+        escrow: BridgeEscrow,
+        // Bridge treasury for mint/burn bridged tokens
+        treasury: BridgeTreasury,
+        pending_messages: Table<BridgeMessageKey, BridgeMessage>,
+        approved_messages: Table<BridgeMessageKey, ApprovedBridgeMessage>,
+        freezed: bool,
+        last_emergency_op_seq_num: u64,
+    }
+
+    // message types
+    const EMERGENCY_OP: u8 = 0;
+    const COMMITTEE_BLOCKLIST: u8 = 1;
+    const COMMITTEE_CHANGE: u8 = 2;
+    const TOKEN: u8 = 3;
+    const NFT: u8 = 4;
+
+    // Emergency Op types
+    const FREEZE: u8 = 0;
+    const UNFREEZE: u8 = 1;
+
+    struct BridgeMessage has copy, store, drop {
+        // 0: token , 1: object ? TBD
+        message_type: u8,
+        message_version: u8,
```

Comment:
What is message_version for?

Reply:
this allow contracts to support different message versions so bridge contracts on different chains can upgrade independently.